src/platform/audio/ffmpegaudiofilecache.cpp

```diff
@@ -55,194 +54,39 @@ FFmpegAudioFileCache::~FFmpegAudioFileCache()
-    }
-
-    open_audio->avio_ctx = avio_alloc_context(buffer, avio_ctx_buffer_size, 0, file, &Read_FFmpeg_Packet, nullptr, nullptr);
-    if (!open_audio->avio_ctx) {
-        captainslog_error("Failed to alloc AVIOContext");
-        Close_FFmpeg_Contexts(open_audio);
-        return false;
-    }
-
-    open_audio->fmt_ctx->pb = open_audio->avio_ctx;
-    open_audio->fmt_ctx->flags |= AVFMT_FLAG_CUSTOM_IO;
-
-    int result = 0;
-    result = avformat_open_input(&open_audio->fmt_ctx, nullptr, nullptr, nullptr);
-    if (result < 0) {
-        char error_buffer[1024];
-        av_strerror(result, error_buffer, sizeof(error_buffer));
-        captainslog_error("Failed 'avformat_open_input': %s", error_buffer);
-        Close_FFmpeg_Contexts(open_audio);
-        return false;
-    }
-
-    result = avformat_find_stream_info(open_audio->fmt_ctx, NULL);
-    if (result < 0) {
-        char error_buffer[1024];
-        av_strerror(result, error_buffer, sizeof(error_buffer));
-        captainslog_error("Failed 'avformat_find_stream_info': %s", error_buffer);
-        Close_FFmpeg_Contexts(open_audio);
-        return false;
-    }
-
-    if (open_audio->fmt_ctx->nb_streams != 1) {
-        captainslog_error("Expected exactly one audio stream per file");
-        Close_FFmpeg_Contexts(open_audio);
-        return false;
-    }
-
-    const AVCodec *input_codec = avcodec_find_decoder(open_audio->fmt_ctx->streams[0]->codecpar->codec_id);
-    if (!input_codec) {
-        captainslog_error("Audio codec not supported: '%u'", open_audio->fmt_ctx->streams[0]->codecpar->codec_tag);
-        Close_FFmpeg_Contexts(open_audio);
-        return false;
-    }
-
-    open_audio->codec_ctx = avcodec_alloc_context3(input_codec);
-    if (!open_audio->codec_ctx) {
-        captainslog_error("Could not allocate codec context");
-        Close_FFmpeg_Contexts(open_audio);
-        return false;
-    }
-
-    result = avcodec_parameters_to_context(open_audio->codec_ctx, open_audio->fmt_ctx->streams[0]->codecpar);
-    if (result < 0) {
-        char error_buffer[1024];
-        av_strerror(result, error_buffer, sizeof(error_buffer));
-        captainslog_error("Failed 'avcodec_parameters_to_context': %s", error_buffer);
-        Close_FFmpeg_Contexts(open_audio);
-        return false;
-    }
-
-    result = avcodec_open2(open_audio->codec_ctx, input_codec, NULL);
-    if (result < 0) {
-        char error_buffer[1024];
-        av_strerror(result, error_buffer, sizeof(error_buffer));
-        captainslog_error("Failed 'avcodec_open2': %s", error_buffer);
-        Close_FFmpeg_Contexts(open_audio);
-        return false;
-    }
-
-    return true;
-}
-
 /**
  * Decode the input data and append it to our wave data stream
  */
 bool FFmpegAudioFileCache::Decode_FFmpeg(FFmpegOpenAudioFile *file)
 {
-    AVPacket *packet = av_packet_alloc();
-    AVFrame *frame = av_frame_alloc();
+    auto on_frame = [](AVFrame *frame, int stream_idx, int stream_type, void *user_data) {
```

Comment:
This will be easier to debug if this is not a lambda function. Just make it a regular function in an anonymous namespace.